tests/test_publish_table.py

```diff
@@ -0,0 +1,361 @@
+import os
+
+import numpy as np
+import pyarrow.parquet as parquet
+import pytest
+
+import audeer
+import audformat
+import audiofile
+
+import audb
+
+
+@pytest.fixture
+def db(tmpdir) -> str:
+    r"""Database fixture.
+
+    Creates a minimal database,
+    containing a media file
+    and a table.
+    The database is not yet stored to disk.
+
+    Args:
+        tmpdir: tmpdir fixture
+
+    Returns:
+        database object, and build dir path
+
+    """
+    name = "test-db"
+    file = "data/file1.wav"
+    table = "files"
+    build_dir = audeer.path(tmpdir, "build")
+    data_dir = audeer.mkdir(build_dir, os.path.dirname(os.path.normpath(file)))
+    audio_file = audeer.path(data_dir, os.path.basename(os.path.normpath(file)))
+    signal = np.zeros((2, 1000))
+    sampling_rate = 8000
+    audiofile.write(audio_file, signal, sampling_rate)
+    db = audformat.Database(name)
+    db.schemes["speaker"] = audformat.Scheme("str")
+    index = audformat.filewise_index([file])
+    db[table] = audformat.Table(index)
+    db[table]["speaker"] = audformat.Column(scheme_id="speaker")
+    db[table]["speaker"].set(["adam"])
+    db.meta["build_dir"] = build_dir
+
+    yield db
+
+
+def expected_table_checksum(path: str) -> str:
+    r"""Expected checksum of table file.
+
+    MD5 sum for CSV table,
+    ``"hash"`` metadata entry for PARQUET files.
+
+    Args:
+        path: path to table file
+
+    """
+    if audeer.file_extension(path) == "parquet":
+        # See https://github.com/audeering/audformat/pull/419
+        return parquet.read_schema(path).metadata[b"hash"].decode()
+    else:
+        return audeer.md5(path)
+
+
+@pytest.mark.parametrize("storage_format", ["csv", "parquet"])
+def test_publish_table_storage_format(db, repository, storage_format):
+    r"""Test publishing and of tables for different storage formats.
+
+    Tables are stored as CSV or PARQUET files.
+    ``audb`` is handling them slightly different:
+
+    * CSV tables are published as ``<table-id>.zip``,
+      and the table checksum is calculated
+      on the original CSV file
+    * PARQUET tables are published as ``<table-id>.parquet``
+      and the table checksum is read
+      from the header of the PARQUET file
+
+    Args:
+        db: db fixture
+        repository: repository fixture,
+            providing a non-persistent repository
+            on a file-system backend
+        storage_format: storage format of tables
+
+    """
+    storage_formats = ["csv", "parquet"]
+    other_storage_format = [sf for sf in storage_formats if sf != storage_format][0]
+
+    build_dir = db.meta["build_dir"]
```

Comment:
These variables are created several times using identical code over this test module.
Would it make sense to turn it into a fixture?

Reply:
Good idea, I added a `build_dir` fixture to that file.

Reply:
I have looked into it. This is good - resolving.